eosmetrics/meson.build

```
@@ -0,0 +1,129 @@
+generated_files = emer_event_recorder_server_dbus_sources
+
+version_config = configuration_data()
+version_config.set('EMER_MAJOR_VERSION', major_version)
+version_config.set('EMER_MINOR_VERSION', minor_version)
+version_config.set('EMER_MICRO_VERSION', micro_version)
+
+generated_files += configure_file(
+  input: 'emtr-version.h.meson',
+  output: 'emtr-version.h',
+  configuration: version_config,
+)
+
+libeosmetrics_public_headers = files(
+  'emtr-apiversion.h',
+  'emtr-aggregate-timer.h',
+  'emtr-enums.h',
+  'emtr-event-recorder.h',
+  'emtr-event-types.h',
+  'emtr-macros.h',
+  'emtr-types.h',
+  'emtr-util.h',
+)
+
+libeosmetrics_sources = files(
+  'emtr-aggregate-timer.c',
+  'emtr-event-recorder.c',
+  'emtr-event-types.c',
+  'emtr-util.c',
+)
+
+libeosmetrics_all_sources = generated_files + libeosmetrics_sources
+
+
+################
+# Dependencies #
+################
+
+libeosmetrics_public_deps = [
+  dependency('gobject-2.0'),
+]
+
+libeosmetrics_private_deps = [
+  dependency('gio-unix-2.0', version: '>= 2.40'),
+  dependency('uuid'),
+]
+
+libeosmetrics_deps = libeosmetrics_public_deps + libeosmetrics_private_deps
+
+
+#################
+# libeosmetrics #
+#################
+
+eosmetrics_export_map_file = 'eosmetrics.map'
+libeosmetrics_library_name = package_name
+
+libeosmetrics = shared_library(
+  libeosmetrics_library_name,
+  libeosmetrics_all_sources,
+  gnu_symbol_visibility: 'default',
+  link_args: [
+    '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), eosmetrics_export_map_file),
+  ],
```

OK, this works, but it's not optimal. Here you compile with default (exported) visibility and then hide all the internal symbols at link time using the linker version script. And that is OK, but it's better to compile with hidden visibility and manually mark APIs that you want to expose using an export macro, so the compiler knows what is exported and what is not and can take that into account when optimizing.

It really doesn't matter very much for a small project like eos-metrics that is unlikely to be used in performance-critical paths, but it's also probably easy to change. Just something to think about....